client/main.lua

```diff
@@ -1,104 +1,110 @@
+        local netId = lib.callback.await('qbx_newsjob:server:spawnVehicle', false, vehicleInfo, coords, Lang:t('info.news_plate')..tostring(math.random(1000, 9999)), true)
+        local timeout = 100
+        while not NetworkDoesEntityExistWithNetworkId(netId) and timeout > 0 do
+            Wait(10)
+            timeout = timeout - 1
+        end
+        local veh = NetToVeh(netId)
+        if veh == 0 then
+            exports.qbx_core:Notify(Lang:t('error.cant_spawn_vehicle'), 'error')
+            return
+        end
         SetVehicleFuelLevel(veh, 100.0)
-        TaskWarpPedIntoVehicle(cache.ped, veh, -1)
-        TriggerEvent("vehiclekeys:client:SetOwner", QBCore.Functions.GetPlate(veh))
         SetVehicleEngineOn(veh, true, true, false)
         SetVehicleLivery(veh, 2)
-        CurrentPlate = QBCore.Functions.GetPlate(veh)
-    end, coords, true)
+        CurrentPlate = GetPlate(veh)
+    end
 end
 
-function MenuGarage()
-    local authorizedVehicles = Config.Vehicles[QBCore.Functions.GetPlayerData().job.grade.level]
+local function menuVehicleGarage()
+    local authorizedVehicles = config.authorizedVehicles[QBX.PlayerData.job.grade.level]
     local optionsMenu = {}
 
     for veh, label in pairs(authorizedVehicles) do
         optionsMenu[#optionsMenu + 1] = {
             title = label,
-            event = 'qb-newsjob:client:TakeOutVehicle',
-            args = {
-                vehicle = veh
-            }
+            onSelect = function()
+                takeOutVehicle(veh)
+            end,
         }
     end
 
     lib.registerContext({
         id = 'weazel_garage_context_menu',
-        title = Lang:t("text.weazel_news_vehicles"),
+        title = Lang:t('info.weazel_news_vehicles'),
         options = optionsMenu
     })
 
     lib.showContext('weazel_garage_context_menu')
 end
 
-function TakeOutHelicopters(vehicleInfo)
-    local coords = Config.Locations["heli"].coords
-    QBCore.Functions.SpawnVehicle(vehicleInfo, function(veh)
-        SetVehicleNumberPlateText(veh, "WZNW"..tostring(math.random(1000, 9999)))
-        SetEntityHeading(veh, coords.w)
+local function takeOutHeli(vehicleInfo)
+    if QBX.PlayerData.job.name == 'reporter' then
+        local coords = config.locations.helicopterStorage.coords.xyz
+        if not coords then
+            local plyCoords = GetEntityCoords(cache.ped)
+            coords = vec4(plyCoords.x, plyCoords.y, plyCoords.z, GetEntityHeading(cache.ped))
+        end
+
+        local netId = lib.callback.await('qbx_newsjob:server:spawnHeli', false, vehicleInfo, coords, Lang:t('info.news_plate')..tostring(math.random(1000, 9999)), true)
+        local timeout = 100
+        while not NetworkDoesEntityExistWithNetworkId(netId) and timeout > 0 do
+            Wait(10)
+            timeout = timeout - 1
+        end
+        local veh = NetToVeh(netId)
+        if veh == 0 then
+            exports.qbx_core:Notify(Lang:t('error.cant_spawn_vehicle'), 'error')
+            return
```

Comment:
How much of this is duplicate code with spawning a vehicle? Consider extracting a function